### PR TITLE
[Ready] Permanent Portal Refactor

### PR DESCRIPTION
**Pull request by Whoneedspacee**
## About The Pull Request

Permanent portals no longer randomly select a portal to use as their exit, and instead stick with the first one they find, they're supposed to only have one linked portal anyways.

Adds a portal exit landmark, one-way portals use this to define their exit as opposed to the weird as hell and pointless portal deletion system I was using before, also nice for mapping since they have different icons so you can more easily tell which is which.

Adds One-Use portals, they're just one-way portals that delete after they're used, was requested for a reference off-station thing someone was making, probably also useful for things like mobs that die and spawn portals to an area, most likely will be used in ice moon.

Goat ruin was the only thing using the permanent portal system for ruins so far since VR hub got removed, so I just moved that over to use the landmarks.

## Why It's Good For The Game

The old code is garbage I wrote like a year ago because it was packaged with VR hub, and someone added copy-pasta at some point, absolutely terrible. This cuts down on a lot.

## Changelog
:cl:
refactor: Permanent Portal fixes and One-Way Portal improvements for mapping.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
very good

**Comment by trexter555** [comment]
does this fix the arena shuttle portals teleporting people into space?

**Comment by Whoneedspacee** [comment]
No because only the goat ruin uses permanent portals. I have no idea how the arena shuttle portals work.

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -185,74 +186,43 @@
 /obj/effect/portal/permanent
 	name = "permanent portal"
 	desc = "An unwavering portal that will never fade."
-	var/id // var edit or set id in map editor
 	hardlinked = FALSE // dont qdel my portal nerd
+	force_teleport = TRUE // force teleports because they're a mapmaker tool
+	var/id // var edit or set id in map editor
 
-/obj/effect/portal/permanent/Initialize(mapload, _lifespan = 0, obj/effect/portal/_linked, automatic_link = FALSE, turf/hard_target_override, atmos_link_override)
```
Comment (spookydonut) [inline code/game/objects/effects/portals.dm]: was this intended to remove initialize?

Comment (Whoneedspacee) [reply code/game/objects/effects/portals.dm]: yes it re-checks links on every teleport now

**Pull request merged**